Sources/BigInt/BigInt.swift

```diff
@@ -0,0 +1,816 @@
+//===--- BigInt.swift -----------------------------------------*- swift -*-===//
+//
+// This source file is part of the Swift Numerics open source project
+//
+// Copyright (c) 2019 - 2020 Apple Inc. and the Swift Numerics project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+public struct BigInt: SignedInteger {
+
+  public typealias Words = [UInt]
+
+  public private(set) var words: Words
```

Comment:
(Implementation tip) For attaswift/BigInt, I found that not allocating a full array for the "small" BigInt cases (up to a couple of digits or so) was a significant overall improvement.